### ISSUE TITLE
FR: mutable-content  in Notifications Composer

### ISSUE THREAD

**Issue by SVyatoslavG** [firebase/firebase-ios-sdk#2863]
### [READ] Step 1: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ make sure the issue title starts with "FR:".
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 10.1
  * Firebase SDK version: 5.20.2
  * Firebase Component: Messaging
  * Component version: 5.20.2

#### Steps to reproduce:

Please add `mutable-content` flag for IOS push notifications in Notifications Composer.
I realize that HTTP Protocol supports `mutable-content` but Notifications Composer allows to target users by App, Version, Country, Language and A/B Testing which hard to do with HTTP Protocol.
I have seen several issues about that in the last year and would like to bring up to the attention that this will be a VERY useful feature.

**Comment by chliangGoogle** [comment]
@SVyatoslavG Thank you for filing the request. Can you provide a bit use case of adding mutable-content flag? Is this for enable service extension on the console? When enable it, what kind of notifications you expect to change when targeting different app. version, etc.

**Comment by SVyatoslavG** [comment]
@chliangGoogle 
Thank you for your response.
I would like to have iOS rich push notifications with media attachments in my app. And yes,  service extension is required to make it work. I want to target different counties, languages and timezones. We have our app translated into 17 languages and  it is required to send them timely push notifications.
Please add this feature into firebase console.

Here is the link on Media about rich push notifications:
https://medium.com/@tsif/ios-10-rich-push-notifications-with-media-attachments-a54dc86586c2

**Comment by chliangGoogle** [comment]
@SVyatoslavG Thank you for your feedback. We are actively working on sth that should resolve your case.

**Comment by gustavogervasio** [comment]
any updates about it?

**Comment by chliangGoogle** [comment]
@gustavogervasio So we recently added image support that will automatically add "mutable_content" in the payload. However if you don't set up an image, it will not have such field.

So I guess this will be a different request and I will forward this to our team and please upvote if anyone finds this feature useful. thanks!

**Comment by gustavogervasio** [comment]
@chliangGoogle I guess it is not working. i tested few minutes ago and i didn't receive the mutable_content when i set the image.
![Screen Shot 2020-04-20 at 15 47 36](https://user-images.githubusercontent.com/3356024/79787970-43531f80-831e-11ea-84fb-cc0bb060fcde.png)

**Comment by chliangGoogle** [comment]
Right, what you requested is a new feature that is not supported yet on the console. What i'm saying is that if you set the message with image, the payload will automatically contains "mutable_content": true in place without you setting it on the console.

**Comment by gustavogervasio** [comment]
thanks for the answer.
Unfortunately i guess firebase won't work for me. Because when i use api i can't send push to segment users and if i use the console i can't send "mutable_content"

**Comment by chliangGoogle** [comment]
Thanks for the feedback, we will keep this open as feature request and bring to our product team.
